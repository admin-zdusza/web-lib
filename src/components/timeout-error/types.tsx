export enum TIMEOUT {
  OCCURED = "TIMEOUT_OCCURED",
  ACKNOWLEDGED = "TIMEOUT_ACKNOWLEDGED",
}

export type TimeoutAction =
  | { type: TIMEOUT.OCCURED }
  | { type: TIMEOUT.ACKNOWLEDGED };
