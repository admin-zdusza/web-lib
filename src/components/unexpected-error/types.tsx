export enum UNEXPECTED_ERROR {
  OCCURED = "UNEXPECTED_ERROR_OCCURED",
  ACKNOWLEDGED = "UNEXPECTED_ERROR_ACKNOWLEDGED",
}

export type UnexpectedErrorAction =
  | { type: UNEXPECTED_ERROR.OCCURED }
  | { type: UNEXPECTED_ERROR.ACKNOWLEDGED };
