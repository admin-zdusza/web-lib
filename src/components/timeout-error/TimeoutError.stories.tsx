import * as React from 'react';
import { Meta, Story } from '@storybook/react/types-6-0';
import { TimeoutError, TimeoutErrorProps } from './TimeoutError';

export default {
  title: 'ClosableTimeoutError',
  component: TimeoutError,
  description: `A timeout error!`,
  argTypes: {},
} as Meta;

const Template: Story<TimeoutErrorProps> = (args) => <TimeoutError {...args}/>;

export const Closable = Template.bind({});
Closable.parameters = {
  viewMode: 'canvas',
  docs: {
    page: null
  }
};

Closable.args = {
  closable: true,
};

export const NonClosable = Template.bind({});
NonClosable.parameters = {
  viewMode: 'canvas',
  docs: {
    page: null
  }
};

NonClosable.args = {
  closable: false,
};
