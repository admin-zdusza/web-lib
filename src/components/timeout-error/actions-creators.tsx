import { TIMEOUT } from "./types";

const timeoutOccured = () => ({ type: TIMEOUT.OCCURED });

const timeoutAcknowledged = () => ({ type: TIMEOUT.ACKNOWLEDGED });

export { timeoutOccured, timeoutAcknowledged };
