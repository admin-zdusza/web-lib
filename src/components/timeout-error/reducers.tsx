import { TIMEOUT, TimeoutAction } from "./types";

const initialState: boolean = false;

const timeoutReducer = (
  state = initialState,
  action: TimeoutAction
): boolean => {
  switch (action.type) {
    case TIMEOUT.OCCURED:
      return true;
    case TIMEOUT.ACKNOWLEDGED:
      return false;
    default:
      return state;
  }
};

export { timeoutReducer };
