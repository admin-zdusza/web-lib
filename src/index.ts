export { Brand } from './components/brand/Brand';
export { TimeoutError } from './components/timeout-error/TimeoutError'
export { timeoutOccured } from './components/timeout-error/actions-creators';
export { timeoutReducer } from './components/timeout-error/reducers';
