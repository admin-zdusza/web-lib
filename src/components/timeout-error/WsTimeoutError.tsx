import React, { Component } from "react";
import Modal from "react-bootstrap/Modal";
import Button from "react-bootstrap/Button";
import { timeoutAcknowledged } from "./actions-creators";

export interface WsTimeoutErrorProps {
  dispatch: Function;
  closable: boolean;
}

const onHide = (closable: boolean, dispatch: Function) => () => {
  if (closable) {
    dispatch(timeoutAcknowledged());
  }
};

const onClick = (dispatch: Function) => () => dispatch(timeoutAcknowledged());

export class WsTimeoutError extends Component<WsTimeoutErrorProps> {
  render() {
    return (
      <Modal
        show={true}
        onHide={onHide(this.props.closable, this.props.dispatch)}
      >
        <Modal.Header closeButton={this.props.closable}>
          <Modal.Title className="text-warning">
            Wolne połączenie internetowe.
          </Modal.Title>
        </Modal.Header>

        <Modal.Body className="text-warning">
          <p>
            Proszę sprawdzić połączenie z internetem i spróbować jeszcze raz.
          </p>
        </Modal.Body>

        {this.props.closable ? (
          <Modal.Footer>
            <Button variant="secondary" onClick={onClick(this.props.dispatch)}>
              Ok
            </Button>
          </Modal.Footer>
        ) : null}
      </Modal>
    );
  }
}
