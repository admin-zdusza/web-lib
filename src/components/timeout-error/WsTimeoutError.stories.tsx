import * as React from 'react';
import { Meta, Story } from '@storybook/react/types-6-0';
import { WsTimeoutError, WsTimeoutErrorProps } from './WsTimeoutError';

export default {
  title: 'ClosableTimeoutError',
  component: WsTimeoutError,
  description: `A timeout error!`,
  argTypes: {},
} as Meta;

const Template: Story<WsTimeoutErrorProps> = (args) => <WsTimeoutError {...args}/>;

export const Closable = Template.bind({});
Closable.parameters = {
  viewMode: 'canvas',
  docs: {
    page: null
  }
};

Closable.args = {
  closable: true,
};

export const NonClosable = Template.bind({});
NonClosable.parameters = {
  viewMode: 'canvas',
  docs: {
    page: null
  }
};

NonClosable.args = {
  closable: false,
};
