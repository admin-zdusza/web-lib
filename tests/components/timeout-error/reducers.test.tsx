import { timeoutReducer } from "../../../src";
import { TIMEOUT } from "../../../src/components/timeout-error/types";

it("should reduce timeout error occured", () => {
  expect(
    timeoutReducer(undefined, { type: TIMEOUT.OCCURED })
  ).toMatchSnapshot();
});

it("should reduce timeout error acknowledged", () => {
  expect(
    timeoutReducer(undefined, { type: TIMEOUT.ACKNOWLEDGED })
  ).toMatchSnapshot();
});

it("should reduce to default", () => {
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  expect(timeoutReducer(false, { type: undefined })).toMatchSnapshot();
});
