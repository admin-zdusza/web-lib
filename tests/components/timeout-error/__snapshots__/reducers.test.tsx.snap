// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should reduce timeout error acknowledged 1`] = `false`;

exports[`should reduce timeout error occured 1`] = `true`;

exports[`should reduce to default 1`] = `false`;
